### ISSUE TITLE
Feature request: demucs_extra_quantized

### ISSUE THREAD

**Issue by lyghter** [facebookresearch/demucs#182]
It would be great if you release "demucs_extra_quantized.th"...

**Comment by adefossez** [comment]
I'm working on training a new model on new extra data, I'll make sure to release that one quantized. This should happen this month or the next.

**Comment by lyghter** [comment]
@adefossez, you are participating in MDX. Will this model be released before the end of the competition?

**Comment by adefossez** [comment]
I don't think I'll get the time to release that before the end of the challenge. If your goal is to save some bandwidth, you can always convert the state to half tensors, which will reduce size by a factor of 2 without impacting performance, although you would need to hack things a bit to apply that to a pre-trained model.

But normally for the MDX challenge you should be fine adding once the pre-trained model to your repo even if it is a bit large no?

**Comment by lyghter** [comment]
Yes, there is no problem with one large model. But I also try many checkpoints of other models. From time to time the size of the repo reaches a certain limit because LFS-files cannot be deleted from repo. Then the submission fails and I have to recreate the repo.

Your competitors on MDX (for example, me) will be happy to receive a new model, even if it is not quantized )

**Comment by adefossez** [comment]
I'll add an option to save models as half precision. This halves model size with no precision loss. Personally I'm using it and deleting old models for each new submission and it works fine. You can also add diffq quantization with the --diffq option, set it to 3e-4 and the model will be super small for no penalty in SDR, but this will make training a bit slower.

I'll try to train a small extra model next week anyway.

**Comment by adefossez** [comment]
I'm not recreating repos simply deleting seems okay. But I do get lots of failures at evaluation (not separation time) and those are a bit annoying. I already contacted the official support for those but it seems the issue persists.

**Comment by adefossez** [comment]
See https://github.com/facebookresearch/demucs/commit/d4b9235eb23a7c9133190f71234f6814ee186418 for the half precision support.

**Comment by adefossez** [comment]
the quantized extra model is training and should be available early next week. Note that this one is trained on a sligthly larger set of extra tracks than the original extra model, so hopefully it should also perform better.

**Comment by adefossez** [comment]
Hey sorry for the lack of update here. As you have guessed the quantized model was not ready in time. The training with extra wav files is a bit unstable, something that I really only discovered during the competitions. With DiffQ on top of it, it wasn't training well. I made training more stable in the next version of Demucs, and that will be necessary to have DiffQ play nicely with it. Now I will release quantized version with extra data only in the next release of Demucs, sometime in september.
